packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/Tasks.js

```diff
@@ -13,10 +13,31 @@ import counterpart from 'counterpart'
 counterpart.registerTranslations('en', en)
 
 function storeMapper (stores) {
-  const { active: classification, demoMode } = stores.classifierStore.classifications
-  const { loadingState } = stores.classifierStore.workflows
-  const { active: step, isThereTaskHelp } = stores.classifierStore.workflowSteps
-  const { loadingState: subjectReadyState } = stores.classifierStore.subjectViewer
+  const {
+    annotatedSteps,
+    classifications,
+    subjectViewer,
+    workflows,
+    workflowSteps
+  } = stores.classifierStore
+  const { active: classification, demoMode } = classifications
+  const { loadingState } = workflows
+  const { active: step, isThereTaskHelp } = workflowSteps
+  const { loadingState: subjectReadyState } = subjectViewer
+  const { latest } = annotatedSteps
+  const readyToClassify = classification && step
+  if (readyToClassify) {
+    const latestSavedStep = latest?.step
+    const stepIsNew = latestSavedStep?.stepKey !== step.stepKey
+    if (stepIsNew) {
+      console.log(`Adding ${step.stepKey} to history`)
+      annotatedSteps.next({
+        step,
+        classification
+      })
+    }
```

Comment:
This is needed to start a new history stack, on the first step for each subject. Maybe there's a better way to do this in the stores themselves?

It may also be needed if we've changed the answer to a branching question ie. the stored history no longer matches our new path through the steps tree.

Reply:
Now handled in the store, using `annotatedSteps.start()` when a new subject loads.